src/main/java/io/supertokens/webserver/api/multitenancy/BaseCreateOrUpdate.java

```diff
@@ -134,8 +215,12 @@ null, null, new JsonObject()
                     tenantConfig.emailPasswordConfig,
                     tenantConfig.thirdPartyConfig,
                     tenantConfig.passwordlessConfig,
-                    firstFactors, tenantConfig.requiredSecondaryFactors, tenantConfig.coreConfig
+                    firstFactors, tenantConfig.useFirstFactorsFromStaticConfigIfEmpty,
+                    tenantConfig.requiredSecondaryFactors, tenantConfig.coreConfig
             );
+            if (useFirstFactorsFromStaticIfEmpty == null) {
+                useFirstFactorsFromStaticIfEmpty = false;
```

Comment:
nope, it will be true if firstFactors is not passed at all. Whenever firstFactors is set, we set this boolean to false. Because this means that user want's to use that value for the tenant.

Reply:
done refactoring